src/validation.cpp

```diff
@@ -1850,7 +1859,7 @@ DisconnectResult CChainState::DisconnectBlock(const CBlock& block, const CBlockI
                 COutPoint out(hash, o);
                 Coin coin;
                 bool is_spent = view.SpendCoin(out, &coin);
-                if (!is_spent || tx.vout[o] != coin.out || pindex->nHeight != coin.nHeight || is_coinbase != coin.fCoinBase) {
+                if (fEnforceBIP30 && (!is_spent || tx.vout[o] != coin.out || pindex->nHeight != coin.nHeight || is_coinbase != coin.fCoinBase)) {
```

Comment:
At first I tried to mimic the behavior in [ConnectBlock](https://github.com/bitcoin/bitcoin/blob/b620b2d58a55a88ad21da70cb2000863ef17b651/src/validation.cpp#L1825-L1834) and the put the `fEnforceBIP30` guard around the entire loop that begins on line 1850, but that would still result in the verification failing later in the process. Moving the guard here allowed the verification process to complete successfully.

I'm assuming that we need to call `SpendCoin` in both cases because we still need to update the view's state even if we ignore the `is_spent` result.